core/base/document.php

```diff
@@ -1425,6 +1426,101 @@ public function __construct( array $data = [] ) {
 		parent::__construct( $data );
 	}
 
+	/**
+	 * Get document settings usage.
+	 *
+	 * @return array
+	 */
+	public function get_usage() {
+		$usage  = [];
+
+		// Ensure not from cache.
+		$document = Plugin::$instance->documents->get( $this->get_id(), false );
+		$controls = $document->get_controls();
+
+		foreach ( $document->get_settings() as $setting_name => $setting_value ) {
+			if ( isset( $controls[ $setting_name ] ) ) {
+				$control = $controls[ $setting_name ];
+				$is_repeater = is_array( $setting_value ) && isset( $control['fields'] );
+
+				if ( ! $this->is_control_default_value( $is_repeater, $controls[ $setting_name ]['default'], $setting_value ) ) {
+					if ( $is_repeater ) {
+						$usage[ $setting_name ] = count( $setting_value );
+					} else {
+						$usage[ $setting_name ] = 1;
+					}
+				}
+			}
+		}
+
+		return $usage;
+	}
+
+	/**
+	 * Get elements usage.
+	 *
+	 * Get the current elements usage by passed elements array parameter.
+	 *
+	 * @throws \Exception
+	 * @return array
+	 */
+	public function get_elements_usage() {
+		$usage = [];
+
+		// Get data manually to avoid conflict with `\Elementor\Core\Base\Document::get_elements_data... convert_to_elementor`.
+		$data = $this->get_json_meta( '_elementor_data' );
+
+		if ( ! $data ) {
+			return $usage;
+		}
+
+		$elements = $this->get_elements_raw_data( $data );
+
+		Plugin::$instance->db->iterate_data( $elements, function ( $element ) use ( &$usage ) {
+			if ( empty( $element['widgetType'] ) ) {
+				$type = $element['elType'];
+				$element_instance = Plugin::$instance->elements_manager->get_element_types( $type );
+			} else {
+				$type = $element['widgetType'];
+				$element_instance = Plugin::$instance->widgets_manager->get_widget_types( $type );
+			}
+
+			if ( ! isset( $usage[ $type ] ) ) {
+				$usage[ $type ] = [
+					'count' => 0,
+					'control_percent' => 0,
+					'controls' => [],
+				];
+			}
+
+			$usage[ $type ]['count']++;
+
+			if ( ! $element_instance ) {
+				return $element;
+			}
+
+			$element_controls = $element_instance->get_controls();
+
+			if ( isset( $element['settings'] ) ) {
+				$settings_controls = $element['settings'];
+				$element_ref = &$usage[ $type ];
+
+				// Add dynamic values.
+				$settings_controls = Usage::add_general_controls( $settings_controls, $element_ref );
+
+				$changed_controls_count = Usage::add_controls( $settings_controls, $element_controls, $element_ref );
```

Comment:
Check why mobile padding didn't appear in the usage data

Reply:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/10234691/169207021-75287991-e964-46d4-9d92-bb8d1b47eb09.png">

Reply:
Please test with a heading that you changed the mobile padding